install/autodetect.go

```diff
@@ -166,5 +168,27 @@ func (k *K8sInstaller) autodetectAndValidate(ctx context.Context) error {
 		return fmt.Errorf("invalid encryption mode")
 	}
 
+	k.autodetectKubeProxy(ctx)
+	return nil
+}
+
+func (k *K8sInstaller) autodetectKubeProxy(ctx context.Context) error {
+	kubeSysNameSpace := "kube-system"
+	dsList, err := k.client.ListDaemonSet(ctx, kubeSysNameSpace, metav1.ListOptions{})
+	if err != nil {
+		k.Log("⏭️ Skipping auto kube-proxy detction")
+		return nil
+	}
+
+	for _, ds := range dsList.Items {
+		if strings.Contains(ds.Name, "kube-proxy") {
+			k.Log("🔮 Auto-detected kube-proxy has been installed")
+			return nil
+		}
+	}
+	k.Log("🔮 Auto-detected kube-proxy has not been installed")
+	k.Log("ℹ️ Cilium will fully replace all functionalities of kube-proxy")
+	// Use HelmOpts to set auto kube-proxy installation
+	k.params.HelmOpts.Values = append(k.params.HelmOpts.Values, "kubeProxyReplacement=strict")
```

Comment:
This needs to set the kube-apiserver endpoint addr too (please see my comment https://github.com/cilium/cilium-cli/issues/1004#issuecomment-1214813920).

Reply:
Thanks for the review. Sorry, I missed setting the API server and API server port. I fixed it and pushed it to the branch. I implemented a new method to get the server’s host and port in [k8s-client](https://github.com/ksankeerth/cilium-cli/blob/dev-1004/k8s/client.go#L126). Please check the changes.